x-pack/legacy/plugins/maps/public/connected_components/map/feature_properties.js

```diff
@@ -110,6 +110,12 @@ export class FeatureProperties extends React.Component {
 
   render() {
 
+    requestAnimationFrame(() => {
```

Comment:
I;m not 100% sure, but this doesn't feel right.

This debouncing on requestAnimationFrame is here, so that we ensure that mapbox does not recalcuate the tooltip until the React renderloop has completed. Which is sometime after `FeatureProperties#render()` has completed.

I don't think we should debounce here in `render()`. It feels like it is the wrong level to bury such a low-level "hack". In a 
 lot of other components, we have moved props-callback and state-updates out of the `render`-method. Also, the `repositionTooltip` handler is really there to trigger a  mapbox-operation, but here we rely on the mounted-state to either make the determination if the state is still valid on the frame-callback. It just looks very tightly coupled.

Rather, I'd suggest to rename the callback to something like `this.props.reevaluatePositionOnNextAnimationFrame`, and call it on `FeatureProperties#componentDidUpdate` (and so no longer `FeatureProperties#render`).

In `mb/view`, handle as you do here, in a `requestAnimationFrame` callback. But rather than check mounted-state (which really isn't relevant at that level, it;s an implementation detail of React), just check if the `tooltipState.location` is valid and not null.